php/commands/site.php

```diff
@@ -1232,71 +1232,41 @@ public function import($args, $assoc_args) {
    * [--instrument=<UUID>]
    * : Change the instrument by setting the ID
    *
+   * @subcommand set-instrument
+   *
    * ## EXAMPLES
    *
-   *  terminus site instrument --site=sitename
+   *  terminus site set-instrument --site=sitename
    */
-  public function instrument($args, $assoc_args) {
+  public function set_instrument($args, $assoc_args) {
     $user        = new User();
-    $instruments = $user->instruments->all();
-    foreach($instruments as $instrument) {
-      $data[$instrument->get('id')] = $instrument->get('label');
+    $instruments = $user->instruments->getMemberList('id', 'label');
+    if (!isset($assoc_args['instrument'])) {
+      $instrument_id = Input::menu(
+        $instruments,
+        null,
+        'Select a payment instrument'
+      );
+    } else {
+      $instrument_id = $assoc_args['instrument'];
+    }
+
+    if (
+      !isset($instruments[$instrument_id])
+      && !in_array($instrument_id, Input::$NULL_INPUTS)
+    ) {
+      $this->logger->error("You do not have permission to attach instrument $instrument_id");
```

Comment:
Actually, Terminus::failure goes to $this->logger->error. I'm slowly trying to rid the app of the sloppy static calls.